src/cgutils.cpp

```diff
@@ -853,6 +853,31 @@ static Value *emit_bounds_check(const jl_cgval_t &ainfo, jl_value_t *ty, Value *
     return im1;
 }
 
+static void emit_inexacterror_unless(Value *cond, jl_value_t *type, jl_value_t *x, jl_codectx_t *ctx)
+{
+    BasicBlock *failBB = BasicBlock::Create(jl_LLVMContext,"fail",ctx->f);
+    BasicBlock *passBB = BasicBlock::Create(jl_LLVMContext,"pass");
+    builder.CreateCondBr(cond, passBB, failBB);
+    builder.SetInsertPoint(failBB);
+    Value *fname_val = stringConstPtr(ctx->funcName);
+#if JL_LLVM_VERSION >= 30700
+    builder.CreateCall(prepare_call(jlinexacterror_func),
+                       { fname_val, literal_pointer_val(type),
+                         boxed(x, ctx, false)});
+#else
+    builder.CreateCall3(prepare_call(jlinexacterror_func),
+                        fname_val, literal_pointer_val(type),
+                        boxed(x, ctx, false));
```

Comment:
Very helpful @yuyichao. Sorry to be dense, but I'm not finding an `auto_unbox(x)`, and `auto_unbox(x, ctx)` seems to return a `Value*` not a `jl_cgval_t`. I'm pushing the current state of the code so you can see more explicitly what I'm trying.